authentication/pipeline/user.py

```diff
@@ -94,18 +93,15 @@ def create_user_via_email(
     if "name" not in data or "password" not in data:
         raise RequirePasswordAndProfileException(backend, current_partial)
 
-    with transaction.atomic():
-        create_user_retval = create_user(
-            strategy, details=details or {}, backend=backend, *args, **kwargs
-        )
-        if not create_user_retval:
-            return
-        user = create_user_retval["user"]
-        user.name = data["name"]
-        user.set_password(data["password"])
-        user.save()
-
-    return {"user": user}
+    return create_user(
+        strategy,
+        details=details or {},
+        backend=backend,
+        name=data["name"],
+        password=data["password"],
+        *args,
+        **kwargs,
+    )
```

Comment:
Nice change. Just wondering how this ends up working. In [social_core.pipeline.user.create_user](https://github.com/python-social-auth/social-core/blob/master/social_core/pipeline/user.py#L68), it seems to be limiting the fields passed to `strategy.create_user` to the members of `USER_FIELDS` (which are `['username', 'email']`). That was the reason I had this whole dance with creating the user then setting the name/pw. I'm guessing the new `SOCIAL_AUTH_EMAIL_USER_FIELDS` setting takes the place of that base `USER_FIELDS` list, but it's unclear to me how/where that happens exactly.

Reply:
Welcome to the automagic of social auth settings`: https://python-social-auth.readthedocs.io/en/latest/configuration/settings.html?highlight=settings

I define it as that just for the email backend so that if we ever add another backend that setting won't stomp on another backend.